src/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -96,17 +98,38 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 						String body = msg.getResponseBody().toString();
 						for (AppPattern p : app.getHtml()) {
 							if (p.getPattern().matcher(body).find()) {
-								logger.debug("WAPP body matched " + app.getName());
+								LOGGER.debug("WAPP body matched " + app.getName());
 								found = true;
 								break;
 							}
 						}
 					}
 					if (! found) {
+						for (Map<String, AppPattern> sp : app.getMetas()) {
+							for (Map.Entry<String, AppPattern> entry : sp.entrySet()) {
+								List<Element> metaElements = source.getAllElements(HTMLElementName.META);
+								if (metaElements != null) {

```

Comment:
Can it be really null? (JavaDoc does not say that it can be null and the code does not return null.)
It could also check that it's not empty.